src/components/StorageOverview/DataResiliency/DataResiliency.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Icon, UtilizationBar } from 'patternfly-react';
+
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '../../Dashboard/DashboardCard';
+import { StorageOverviewContext } from '../StorageOverviewContext';
+import { InlineLoading } from '../../Loading';
+import { getDataResiliencyData } from '../../../selectors/prometheus/selectors';
+
+const DataResiliencyStatusBody = ({ status }) =>
+  status ? (
+    <React.Fragment>
+      <div className="kubevirt-data-resiliency__status-title--ok">Your Data Is Resilient</div>
+      <div className="kubevirt-data-resiliency__icon--ok">
+        <Icon type="fa" name="check-circle" size="3x" />
+      </div>
+    </React.Fragment>
+  ) : (
+    <React.Fragment>
+      <div className="kubevirt-data-resiliency__icon--error">
+        <Icon type="fa" name="exclamation-triangle" size="1x" />
+      </div>
+      <div className="kubevirt-data-resiliency__status-title--error">Some Error Occurred</div>
+    </React.Fragment>
+  );
+
+const DataResiliencyBuildBody = ({ progressPercentage }) => (
+  <React.Fragment>
+    <div className="kubevirt-data-resiliency__title">Rebuilding data resiliency</div>
+    <UtilizationBar
+      now={progressPercentage}
+      description="Rebuilding in Progress"
+      descriptionPlacementTop
+      label={`${progressPercentage}%`}
+    />
+  </React.Fragment>
+);
+
+export const DataResiliency = ({ activePgRaw, totalPgRaw }) => {
+  const activePg = getDataResiliencyData(activePgRaw);
+  const totalPg = getDataResiliencyData(totalPgRaw);
+  const progressPercentage = (activePg / totalPg) * 100;
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Data Resiliency</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody
+        className="kubevirt-data-resiliency__dashboard-body"
+        isLoading={!(activePgRaw && totalPgRaw)}
+        LoadingComponent={InlineLoading}
```

Comment:
The reason for passing the `LoadingComponent` down is the ability to reuse two different implementations.
The `web-ui` recently uses non-patternfly implementation. For consistency, its use has priority.
Can you please pass the component down from the web-ui? Please have a look at `ClusterOverview` for how to.
The `InlindeLoading` is a default and so should be set within `defaultProps`.

FYI: once openshift/console addopts patternfly's component, we will get rid of this passing.

Reply:
done ..thanx for the information